crates/typst-cli/src/compile.rs

```diff
@@ -335,6 +354,55 @@ impl ExportCache {
     }
 }
 
+/// Writes a Makefile rule describing the relationship between `output` and
+/// `dependencies` to `makefile_deps_path`.
+fn create_makefile_deps(
+    makefile_deps_path: &Path,
+    output: &Path,
+    dependencies: impl Iterator<Item = PathBuf>,
+) -> io::Result<()> {
+    // Based on `munge` in libcpp/mkdeps.cc from the GCC source code. This isn't
+    // perfect as some special characters can't be escaped.
+    fn munge(p: &Path) -> String {
+        let s = p.to_string_lossy();
+        let mut res = String::with_capacity(s.len());
+        let mut slashes = 0;
+        for c in s.chars() {
+            match c {
+                '\\' => slashes += 1,
+                '$' => {
+                    res.push('$');
+                    slashes = 0;
+                }
+                ' ' | '\t' => {
+                    // `munge`'s source contains a comment here that says: "A
+                    // space or tab preceded by 2N+1 backslashes represents N
+                    // backslashes followed by space..."
+                    res.push_str(&"\\".repeat(slashes + 1));
```

Comment:
I'd use a for loop here rather than allocate a temporary string.

Reply:
Ah good idea, I've fixed that in 60be0c96f4217e7738d84ac1f070da1a3969097f.